kategory-core/src/test/kotlin/kategory/data/ValidatedTest.kt

```diff
@@ -165,27 +170,65 @@ class ValidatedTest : UnitSpec() {
         }
 
         "Cartesian builder should build products over homogeneous Validated" {
-            Validated.applicative(StringMonoidInstance).map(
+            Validated.applicative<String>().map(
                     Valid("11th"),
                     Valid("Doctor"),
                     Valid("Who"),
                     { (a, b, c) -> "$a $b $c" }) shouldBe Valid("11th Doctor Who")
         }
 
         "Cartesian builder should build products over heterogeneous Validated" {
-            Validated.applicative(StringMonoidInstance).map(
+            Validated.applicative<String>().map(
                     Valid(13),
                     Valid("Doctor"),
                     Valid(false),
                     { (a, b, c) -> "${a}th $b is $c" }) shouldBe Valid("13th Doctor is false")
         }
 
         "Cartesian builder should build products over Invalid Validated" {
-            Validated.applicative(StringMonoidInstance).map(
+            Validated.applicative<String>().map(
                     Invalid("fail1"),
                     Invalid("fail2"),
                     Valid("Who"),
                     { "success!" }) shouldBe Invalid("fail1fail2")
         }
+
+        "CombineK should combine Valid Validated" {
+            val valid = Valid("Who")
+
+            Validated.semigroupK<String>().combineK(valid, valid) shouldBe(Valid("Who"))
+        }
+
+        "CombineK should combine Valid and Invalid Validated" {
+            val valid = Valid("Who")
+            val invalid = Invalid("Nope")
+
+            Validated.semigroupK<String>().combineK(valid, invalid) shouldBe(Valid("Who"))
+        }
+
+        "CombineK should combine Invalid Validated" {
+            val invalid = Invalid("Nope")
+
+            Validated.semigroupK<String>().combineK(invalid, invalid) shouldBe(Invalid("NopeNope"))
+        }
+
+        "Combine should combine Valid Validated" {
+            val valid: Validated<String, String> = Valid("Who")
+
+            valid.combine(valid) shouldBe(Valid("WhoWho"))
+        }
+
+        "Combine should combine Valid and Invalid Validated" {
+            val valid = Valid("Who")
+            val invalid = Invalid("Nope")
+
+            valid.combine(invalid) shouldBe(Invalid("Nope"))
+        }
+
+        "Combine should combine Invalid Validated" {
+            val invalid: Validated<String, String> = Invalid("Nope")
```

Comment:
@kategory/maintainers One thing I couldn't avoid is to specify String here (or any other inplicit Semigroup that we have, like Int), as Nothing wouldn't work with the reified needed for the `combine` call with "implicit" Semigroups and Any or Unit are not provided global instances (shall we maybe create them and return something dummy there? 🤔 )

Reply:
This is fine AFAIK since it's in tests